docs/how_tos/feedback.rst

```diff
@@ -0,0 +1,409 @@
+.. _feedback/data/sagas.js: https://github.com/openedx/frontend-app-payment/blob/master/src/feedback/data/sagas.js
+
+.. _List of messages:
+
+Format: List of messages
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: json
+
+    {
+        "messages": [
+            {
+                "data": "",
+                "code": "",
+                "message_type": "",
+                "user_message": ""
+            },
+            {
+                "data": "",
+                "code": "",
+                "message_type": "",
+                "user_message": ""
+            },
+            <...>
+        ]
+    }
+
+Send this format with a HTTP code for a `successful response (2xx)`_.
+
+.. _successful response (2xx): https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#successful_responses
+
+.. _List of errors:
+
+Format: List of errors
+~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: json
+
+    {
+        "errors": [
+            {
+                "data": "",
+                "error_code": "",
+                "message_type": "",
+                "user_message": ""
+            },
+            {
+                "data": "",
+                "error_code": "",
+                "message_type": "",
+                "user_message": ""
+            },
+            <...>
+        ]
+    }
+
+This format is identical to the `List of messages`_, but uses ``error_code``
+instead of ``code``.
+
+.. _Dictionary of field errors:
+
+Format: Dictionary of field errors
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: json
+
+    {
+        "field_errors": {
+            "<field_name_1>": {
+                "error_code": "",
+                "user_message": ""
+            },
+            "<field_name_2>": {
+                "error_code": "",
+                "user_message": ""
+            },
+            <...>
+        }
+    }
+
+``<field_name_1>`` and ``<field_name_2>`` are placeholders for the `name`_ of
+the `Redux Form Field`_ the error message should be placed next to.
+
+The handling of ``field_errors`` is done by `Redux Form's stopSubmit`_ in
+`handleReduxFormValidationErrors()`_. ``handleReduxFormValidationErrors()`` transforms
+``field_errors`` into the format required by `Redux Form's SubmissionError`_.
+
+.. _name: https://redux-form.com/8.3.0/docs/api/field.md/#-code-name-string-code-required-
+.. _Redux Form's stopSubmit: https://redux-form.com/8.3.0/docs/api/actioncreators.md/#-code-stopsubmit-form-string-errors-object-code-
+.. _Redux Form's SubmissionError: https://redux-form.com/8.3.0/docs/api/submissionerror.md/
+.. _handleReduxFormValidationErrors(): https://github.com/openedx/frontend-app-payment/blob/1d631c51035eb8405ce9b03e0fa64a5a6e386268/src/payment/data/sagas.js#L59-L64
+
+.. _Single error:
+
+Format: Single error
+~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: json
+
+    {
+        "error_code": "",
+        "user_message": ""
+    }
+
+.. note::
+
+    For reasons unknown, this format undergoes some `custom handling`_
```

Comment:
I think this is because we are creating errors in the checkout for [stripe](https://github.com/openedx/frontend-app-payment/blob/1d631c51035eb8405ce9b03e0fa64a5a6e386268/src/payment/payment-methods/stripe/service.js#L96) and [cybersource](https://github.com/openedx/frontend-app-payment/blob/1d631c51035eb8405ce9b03e0fa64a5a6e386268/src/payment/payment-methods/cybersource/service.js#L154) (theres also one in apply pay) but not wrapping them in anything so they would always be a [fallback message ](https://github.com/openedx/frontend-app-payment/blob/1d631c51035eb8405ce9b03e0fa64a5a6e386268/src/feedback/data/sagas.js#L16) without this. havent looked too deep into this as to why we are creating the error there though

Reply:
Curious! Those two lines in question pre-date Stripe, though.